dockstore-webservice/src/main/resources/openapi3/openapi.yaml

```diff
@@ -4201,6 +4201,95 @@ paths:
           description: default response
       tags:
       - users
+  /users/{userId}/workflows:
+    get:
+      description: List all workflows owned by the authenticated user.
+      operationId: userWorkflows
+      parameters:
+      - description: User ID
+        in: path
+        name: userId
+        required: true
+        schema:
+          format: int64
+          type: integer
+      responses:
+        default:
+          content:
+            application/json:
+              schema:
+                items:
+                  $ref: '#/components/schemas/Workflow'
+                type: array
+          description: default response
+      security:
+      - bearer: []
+      tags:
+      - users
+    patch:
+      description: Adds the logged-in user to any Dockstore workflows that they should
+        have access to.
+      operationId: addUserToDockstoreWorkflows
+      parameters:
+      - description: User to update
+        in: path
+        name: userId
+        required: true
+        schema:
+          format: int64
+          type: integer
+      requestBody:
+        content:
+          '*/*':
+            schema:
+              type: string
+        description: This is here to appease Swagger. It requires PATCH methods to
```

Comment:
Ugh, this is a sign it should be POST (and not PUT or PATCH).  Do we have a PATCH without a body before?

Reply:
Not that I know of. @denis-yuen @coverbeck  What do you think? Would like to get this merged so that I can get the UI passing.

Reply:
I think it's still idempotent, so not a POST.   i.e. "Note that while idempotent operations produce the same result on the server (no side effects), the response itself may not be the same (e.g. a resource's state may change between requests)." (I think GitHub changes are more akin to changes in a resource's state than in the entity that we are sending ... or not sending in our case).
https://www.restapitutorial.com/lessons/idempotency.html

That said, I'm well beyond caring at this point. I'd say merge it.

Reply:
I would argue that it is changing server state, and should be a POST.

But also agree it's not worth hair-splitting.

Reply:
Idempotent is not what's defining between POST and PUT, but PUT definitely needs to be idempotent.  Most of our GET requests are also idemptotent, but we wouldn't use PUT for example.  The fact that swagger complains about no body is definite issue.

https://tools.ietf.org/html/rfc7231#section-4.3.4 

PUT is meant to replace the existing resource with a mesage payload (the body), having no body means it's definitely incorrect.

Reply:
It seems to me that you could hit the endpoint multiple times, but it is just going to end up in the same state, our DB being up-to-date with the current information from GitHub. 

That GitHub can change doesn't seem enough to me to make it non-idempotent. 
https://dev.dockstore.net/api/metadata/okHttpCachePerformance is going to change all the time depending on how our cache is performing, so would a GET request that gathers, say, statistics on the current number of database connections open even though that can change.

Reply:
My feeling is because it can trigger a state change on the server, even if said state ultimately comes from GitHub, that it's not idempotent.  I guess that's where we differ.

In the cache example, the GET itself is not changing the state, so I think that's fine.

It's merged anyway, so it's all moot. :)